### PR TITLE
openharmony application switch from FA model to stage model

### PR DESCRIPTION
**Pull request by qiuguohua**
https://docs.openharmony.cn/pages/v3.2Beta/zh-cn/application-dev/ability/ability-brief.md/

Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: scripts/native-pack-tool/source/platforms/openharmony.ts [hunk 1]
```diff
@@ -61,14 +61,32 @@ export class OpenHarmonyPackTool extends NativePackTool {
             console.log(`rewrite buildCfgJson err: ${e}`);
         }
 
+        const moduleFile = ps.join(ohosProjDir, 'entry/src/main/module.json5');
+        const moduleJSON = fs.readJSONSync(moduleFile);
```
Comment (oahcy) [inline scripts/native-pack-tool/source/platforms/openharmony.ts]: json5 file use JSON5.parse better。

File: scripts/native-pack-tool/source/platforms/openharmony.ts [hunk 2]
```diff
@@ -61,14 +61,32 @@ export class OpenHarmonyPackTool extends NativePackTool {
...
+        const cfg = this.params.platformParams.orientation;
```
Comment (oahcy) [inline scripts/native-pack-tool/source/platforms/openharmony.ts]: platformParams.cfg

File: scripts/native-pack-tool/source/platforms/openharmony.ts [hunk 3]
```diff
@@ -61,14 +61,32 @@ export class OpenHarmonyPackTool extends NativePackTool {
...
+        // copy jsb-adapter to entry/src/main/ets/cocos/jsb-adapter
         const mainDir = ps.join(ohosProjDir, 'entry/src/main');
-        await cchelper.copyFileSync("", ps.join(assetsDir, 'jsb-adapter/engine-adapter.js'), "", ps.join(mainDir, 'ets/MainAbility/cocos/jsb-adapter/engine-adapter.js'));
-        await cchelper.copyFileSync("", ps.join(assetsDir, 'jsb-adapter/web-adapter.js'), "", ps.join(mainDir, 'ets/MainAbility/cocos/jsb-adapter/web-adapter.js'));
+        await cchelper.copyFileSync("", ps.join(assetsDir, 'jsb-adapter/engine-adapter.js'), "", ps.join(mainDir, 'ets/cocos/jsb-adapter/engine-adapter.js'));
+        await cchelper.copyFileSync("", ps.join(assetsDir, 'jsb-adapter/web-adapter.js'), "", ps.join(mainDir, 'ets/cocos/jsb-adapter/web-adapter.js'));
 
-        const cfgFile = ps.join(ohosProjDir, 'entry/src/main/config.json');
+        const cfgFile = ps.join(ohosProjDir, 'AppScope/app.json5');
         const configJSON = fs.readJSONSync(cfgFile);
```
Comment (oahcy) [inline scripts/native-pack-tool/source/platforms/openharmony.ts]: json5 file use JSON5.parse

File: scripts/native-pack-tool/source/platforms/openharmony.ts [hunk 4]
```diff
@@ -116,8 +134,10 @@ export class OpenHarmonyPackTool extends NativePackTool {
         const hdcExe = "hdc_std";
         const projectDir = this.paths.platformTemplateDirInPrj;
         const packageName = this.params.platformParams.packageName;
-        const configJson = fs.readJSONSync(ps.join(projectDir, 'entry/src/main/config.json'));
-        const moduleId = configJson.module.package + configJson.module.mainAbility;
+        const configJson = fs.readJSONSync(ps.join(projectDir, 'AppScope/app.json5'));
```
Comment (oahcy) [inline scripts/native-pack-tool/source/platforms/openharmony.ts]: json5 file

File: scripts/native-pack-tool/source/platforms/openharmony.ts (hunk 1 above)
Comment (qiuguohua) [reply scripts/native-pack-tool/source/platforms/openharmony.ts]: OK

File: scripts/native-pack-tool/source/platforms/openharmony.ts (hunk 3 above)
Comment (qiuguohua) [reply scripts/native-pack-tool/source/platforms/openharmony.ts]: OK

File: scripts/native-pack-tool/source/platforms/openharmony.ts (hunk 4 above)
Comment (qiuguohua) [reply scripts/native-pack-tool/source/platforms/openharmony.ts]: OK

File: scripts/native-pack-tool/source/platforms/openharmony.ts (hunk 2 above)
Comment (qiuguohua) [reply scripts/native-pack-tool/source/platforms/openharmony.ts]: ok

**Review by minggo — APPROVED** [review]

**Review by oahcy — APPROVED** [review]

**Pull request merged**